Update screenshots and wording for the Aura GraphQL create flow (#303)
neo4j/docs-graphql: modules/ROOT/pages/aura-graphql/api-creation.adoc

== Console

. In the Aura Console, select **Data APIs** from the left side navigation.
. Ensure you are connected to the Neo4j AuraDB instance that the GraphQL API will be used with.
. Select **Define my own** to write your own type definitions or **Fetch from instance** to generate type definitions based on the existing data in your AuraDB.
. Type definitions
+
This is where you describe the graph database in the AuraDB that the GraphQL API will be used with.
The type definitions follow the rules of the Neo4j GraphQL Library with the exception that custom resolvers cannot be used.
+
If you already have data in the AuraDB, a quick way to obtain type definitions is to use the link:https://graphql-toolbox.neo4j.io[Neo4j GraphQL Toolbox].
The Neo4j GraphQL Toolbox has the ability to connect to an AuraDB, automatically create type definitions and allow GraphQL operations.
+

> Alternatively you can write your own type definitions from first principles by following the guidance provided in xref:index.adoc[Neo4j GraphQL Library] documentation.

Reviewer: ```suggestion
Alternatively you can write your own type definitions from scratch by following the guidance provided in xref:index.adoc[Neo4j GraphQL Library] documentation.
```

https://en.wikipedia.org/wiki/First_principle
fancy ;)

== Console

. In the Aura Console, select **Data APIs** from the left side navigation.

> . Ensure you are connected to the Neo4j AuraDB instance that the GraphQL API will be used with.

Reviewer: ```suggestion
. Ensure you are connected to the Neo4j AuraDB instance with which you want to use the GraphQL API. The **Instance** drop-down at the top left shows your current connection.
```
not quite sure about the "where" (was trying to avoid "that you want to use ... with"
what does the native speaker say?

Author: Probably:

> Ensure you are connected to the Neo4j AuraDB instance with which you want to use the GraphQL API.

Reviewer: updated the suggestion

+

> image::aura-graphql/create/connected-instance.png[]

Reviewer: this screenshot has lots of "white space" to the right that could be cut

in general though - is it really necessary?
from a maintenance point of view, we are trying to keep screenshots to a minimum and only add them when they add value. instead of showing this screenshot in particular, we could add a line that says "The **Instance** drop down at the top left shows your current connection" or something similar

Author: I'm all for removing screenshots :+1: I only added them because it seemed to be inline with the current docuemntation.

+

> . Select either to generate type definitions based on the existing data in your AuraDB or to write your own type definitions.

Reviewer: ```suggestion
. Select **Define my own** to write your own type definitions or **Fetch from instance** to generate type definitions based on the existing data in your AuraDB.
```

+

> image::aura-graphql/create/type-definitions-choice.png[]

Reviewer: lots of space here, i think it could be cropped down, however do we need the screenshot? similar arguments as above

+
. Type definitions
+
This is where you describe the graph database in the AuraDB that the GraphQL API will be used with.

> The type definitions are the same as those used with Neo4j GraphQL Library with the exception that custom resolvers cannot be used.

Reviewer: ```suggestion
The type definitions follow the rules of the Neo4j GraphQL Library with the exception that custom resolvers cannot be used.
```

+

> If you already have data in the AuraDB, a quick way to obtain type definitions is to use the https://graphql-toolbox.neo4j.io[Neo4j GraphQL Toolbox]. This facility has the ability to connect to an AuraDB, automatically create type definitions and allow GraphQL operations.

Reviewer: ```suggestion
If you already have data in the AuraDB, a quick way to obtain type definitions is to use the link:https://graphql-toolbox.neo4j.io[Neo4j GraphQL Toolbox].
The Neo4j GraphQL Toolbox has the ability to connect to an AuraDB, automatically create type definitions and allow GraphQL operations.
```

+

> Alternatively you can write your own type definitions from first principles by following the guidance provided in xref:index.adoc[Neo4j GraphQL Library] documentation.

Reviewer: what does "from first principles" mean?

Author: This was preexisting text just moved around in this PR, I don't know why it was written that way before :sweat_smile:

+
If you are using GraphQL Federation then make sure to select **Enable GraphQL subgraph**.
+

> image::aura-graphql/create/type-definitions-refinement.png[]

Reviewer: this is to show "Enable GraphQL subgraph", correct?
i think this screenshot could also be omitted

+

> . Once you are satisfied with the type definitions, select **Try out the API** to proceed to the next step where you can simulate GraphQL queries and mutations against the type definitions you have provided. Note that this while this does run against your database, it does not create a GraphQL API that can be used by applications until you have completed the final step of the creation process.

Reviewer: ```suggestion
. Once you are satisfied with the type definitions, select **Try out the API** to proceed. The next step lets you simulate GraphQL queries and mutations against the type definitions you have provided. Note that while this does run against your database, it does not create a GraphQL API that can be used by applications until you have completed the final step of the creation process.
```

+
image::aura-graphql/create/query-editor.png[]
+

> . Proceed to get API ready and fill in the details in the final step.

Reviewer: ```suggestion
. Proceed to get the API ready and fill in the details in the final step.
```

+
At this point the configuration is complete for the GraphQL API.
Select **Create** to proceed and create the GraphQL API.

> Alternatively, go **Back** to previous steps to make changes, or simply navigate away to exit without creating the API.

Reviewer: ```suggestion
Alternatively, go **Back** to previous steps to make changes, or navigate away to exit without creating the API.
```
what's "navigate away" exactly?

Author: "navigate away" as browser terminology which I thought was colloquial when describing "clicking to another part of the website or leaving the website entirely"
